microsoft/testsuites/stress/tlb_stress.py

```diff
@@ -0,0 +1,960 @@
+
+        return exists
+
+    def _unlock_perf_counters(self) -> None:
+        """Unlock perf counters so dTLB/iTLB/context-switch counters aren't zero"""
+        try:
+            # Check current setting
+            result = self.node.execute(
+                "cat /proc/sys/kernel/perf_event_paranoid", no_error_log=True
+            )
+            current_value = result.stdout.strip()
+
+            if current_value != "-1":
+                self.node.execute("sysctl -w kernel.perf_event_paranoid=-1", sudo=True)
+        except Exception as e:
+            self._log.warning(
+                f"Failed to unlock perf counters: {e}. Will use sudo for perf commands."
+            )
+
+    def _probe_perf_events(self) -> Dict[str, bool]:
+        """Probe which perf events are supported on this system"""
+        event_support = {}
+
+        # Test events to probe
+        events_to_test = [
+            "dTLB-load-misses",
+            "iTLB-load-misses",
+            "context-switches",
+            "cpu-migrations",
+            "page-faults",
+            "cycles",
+            "instructions",
+            "cache-misses",
+        ]
+
+        for event in events_to_test:
+            try:
+                # Quick test: perf stat -e EVENT -- /bin/true
+                result = self.node.execute(
+                    f"perf stat -e {event} -- /bin/true",
+                    sudo=True,
+                    no_error_log=True,
+                    timeout=10,
+                )
+                # If no error and doesn't contain "<not supported>", event is supported
+                supported = (
+                    result.exit_code == 0
+                    and "<not supported>" not in result.stderr
+                    and "not supported" not in result.stderr
+                )
+                event_support[event] = supported
+            except Exception:
+                event_support[event] = False
+
+        return event_support
+
+    def _get_fallback_events(self, event_support: Dict[str, bool]) -> str:
+        """Get fallback event list based on what's supported"""
+        # Always include generic software events
+        base_events = []
+
+        # Add supported hardware events
+        if event_support.get("context-switches", False):
+            base_events.append("context-switches")
+        if event_support.get("cpu-migrations", False):
+            base_events.append("cpu-migrations")
+        if event_support.get("page-faults", False):
+            base_events.append("page-faults")
+
+        # Add PMU events if supported
+        if event_support.get("dTLB-load-misses", False):
+            base_events.append("dTLB-load-misses")
+        if event_support.get("iTLB-load-misses", False):
+            base_events.append("iTLB-load-misses")
+
+        # If no events supported, use minimal set
+        if not base_events:
+            base_events = ["context-switches", "page-faults"]
+            self._log.warning(
```

Comment:
Don't print warning level. move it to info or debug level.

Reply:
moved to debug.